bm-instance-agent/bm_instance_agent/systems/linux/distro/kylin.py

```
@@ -30,6 +41,105 @@ def detach_port(self, instance_obj, network_obj):
+        for attempt in range(3):
+            try:
+                with open(slaves_file_path, 'r') as f:
+                    slaves = f.read().strip()
+
+                if slaves:
+                    LOG.info("the slaves file is %s", slaves)
+                    return ','.join(slaves.split())
+
+                LOG.warning("Attempt %d/%d: Slaves file is empty. Retrying...", attempt + 1, max_retries)
+            except FileNotFoundError:
+                LOG.error("Attempt %d/%d: Slaves file not found. Retrying...", attempt + 1, max_retries)
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)